### ISSUE TITLE
Clojurescript Watch

### ISSUE THREAD

**Issue by whamtet** [clojurephant/clojurephant#157]
## Desired Behavior

Is there any support for clojurescript watch (instead of build)?  Happy to submit a PR if you point out which classes need changing.

**Comment by ajoberstar** [comment]
Sorry about the close, I fat fingered my phone.

Gradle has a built in continuous mode. Add `-t` to your Gradle command. This won't work if you're running the reply, but can at least keep building on changes.

Let me know if there are additional missing capabilities to what you're looking for.

**Comment by whamtet** [comment]
Thanks for the quick reply Andrew,

Your Gradle plugin calls the clojurescript compiler.build command.  There
is also compiler.watch command which should be more performant.  By the
looks of things, you're forking a separate JVM to run each compilation, is
that correct?  If you point out the key classes I'm happy to submit a PR.

**Comment by ajoberstar** [comment]
ClojureScript builds happen through a combination of [these plugin](https://github.com/clojurephant/clojurephant/tree/main/clojurephant-plugin/src/main/java/dev/clojurephant/plugin/clojurescript) classes and [this tools](https://github.com/clojurephant/clojurephant/tree/main/clojurephant-plugin/src/main/java/dev/clojurephant/plugin/clojurescript) namespace.

I think this one needs a little thought still about how the experience should work in combination with Gradle.

**Comment by ajoberstar** [comment]
e.g. This might be a case to enhance the ClojureScriptCompile task to support the incremental task API, such that the continuous updates would feed just the changed files in. Essentially, we'd let Gradle handle most of the machinery that `cljs.build/watch`  does (it's mostly handling file watching and then re-running `cljs.build/build`).

The reason to do this over directly calling `cljs.build/watch` is that it allows other tasks before (or after) the ClojureScript build to also run on any changes and ensures Gradle fully understands the inputs/outputs (for up-to-date checking).

**Comment by whamtet** [comment]
Looks like that would require a PR to the clojurescript project.

**Comment by ajoberstar** [comment]
I don't think it would require changes to ClojureScript itself; they expose all the APIs we would need. From their side, we'd mostly need just `cljs.build/build` (and maybe a couple of the `cljs.analyzer` calls to get from a source file to the ns and to the output file). Gradle can already watch for file changes and can provide the list of changed files to the task. We'd just need the code within the plugin to do this.
